client/jetpack-app/plans/main.tsx

```diff
@@ -118,6 +118,7 @@ const JetpackAppPlans: React.FC< JetpackAppPlansProps > = ( { paidDomainName, or
 						hidePlanTypeSelector
 						hidePlansFeatureComparison
 						setSiteUrlAsFreeDomainSuggestion={ setSiteUrlAsFreeDomainSuggestion }
+						flowName="onboarding-jetpack-app"
```

Comment:
Since technically there is no `flow` involved  here, what you can do is use the `intent` property above  instead.

Just feed the intent into the `useModalResolutionCallback` hook and check that, also make sure to write a unit test at 👍 

https://github.com/Automattic/wp-calypso/blob/ad35534162a9a39f823b260d4437550bee614ce6/client/my-sites/plans-features-main/components/plan-upsell-modal/test/plan-upsell-modal.test.tsx